### PR TITLE
Get mdns-publisher via multi-stage build

### PR DESCRIPTION
**Pull request by techiebod**
Removed pkgs and pip to thin this down much further now, also note I snuck in a change to pin us to a less specific version of Python

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -50,8 +47,6 @@ COPY cname.py .
 COPY restart.sh .
 RUN touch ./cnames
 
-RUN pip install mdns-publisher
```
Comment (hardillb) [inline Dockerfile:53]: I've just tested this locally and it fails with this line missing with the following:

```
forego      | starting cname.1 on port 603400
cname.1     | Traceback (most recent call last):
cname.1     |   File "/usr/src/app/cname.py", line 15, in <module>
cname.1     |     from mpublisher import AvahiPublisher
cnforego      | starting cname.1 on port 603400
cname.1     | Traceback (most recent call last):
cname.1     |   File "/usr/src/app/cname.py", line 15, in <module>
cname.1     |     from mpublisher import AvahiPublisher
cname.1     | ModuleNotFoundError: No module named 'mpublisher'
ame.1     | ModuleNotFoundError: No module named 'mpublisher'
```

We definitely need to install `mdns-publisher`

Comment (hardillb) [reply Dockerfile:53]: Now I think about it, we need the build tools because the mdns-publisher has native components so they are needed to build it.

Comment (techiebod) [reply Dockerfile:53]: Ahh sorry, must have fallen victim to the build cache, as rebuilding it again with --no-cache I get the same issue.  I guess the only other option then is to create a builder docker container for it and then copy across.  May just go do that for educational purposes, but I doubt the complexity is worth the ~200MB saved from the image.  But once it's done I'll come back with more info

**Comment by techiebod** [comment]
Saves a little more disk space:
```
$ docker image ls -f 'reference=*/nginx-proxy-avahi-helper'
REPOSITORY                           TAG       IMAGE ID       CREATED         SIZE
techiebod/nginx-proxy-avahi-helper   latest    902838d0d51b   7 minutes ago   188MB
hardillb/nginx-proxy-avahi-helper    latest    1439858f1045   2 days ago      564MB
```
And makes the Python version a variable.  Sorry for back and forth!

Tested with a no-cache build this time! And sorry for the odd ARG name, but PYTHON_VERSION is in the image as an Env which appears to take precedence over ARGs, if there's a better way to do this please let me know

File: Dockerfile [hunk 2]
```diff
@@ -32,26 +33,28 @@ RUN git clone https://github.com/nginx-proxy/forego/ \
    && cd - \
    && rm -rf /go/forego
 
+FROM python:${PYTHON_VER} as pythonbuilder
+RUN apt-get update && apt-get install -y build-essential libdbus-1-dev
```
Comment (hardillb) [inline Dockerfile]:
```suggestion
RUN apt-get update && apt-get install -y build-essential libdbus-1-dev cmake
```

Comment (techiebod) [reply Dockerfile]: Sure, but why do we need cmake?

Comment (hardillb) [reply Dockerfile]: To build ninja... to build the native components of mdns-publisher on ARMv7 (where there are no pre-built binaries)

**Pull request merged**